scrapyd_k8s/object_storage/libcloud_driver.py

```diff
@@ -0,0 +1,188 @@
+                raise ValueError(f"Environment variable '{env_var}' is not set for argument '{arg_name}'.")
+
+        result = self.VARIABLE_PATTERN.sub(replace_var, value)
+        result = result.replace(r'\${', '${')
+        return result
+
+    def upload_file(self, local_path):
+        """
+        Uploads a file to the object storage container.
+
+        Parameters
+        ----------
+        local_path : str
+            The local file path of the file to be uploaded.
+
+        Returns
+        -------
+        None
+
+        Logs
+        ----
+        Logs information about the upload status or errors encountered.
+        """
+        object_name = os.path.basename(local_path)
+        try:
+            container = self.driver.get_container(container_name=self._container_name)
+            self.driver.upload_object(
+                local_path,
+                container,
+                object_name,
+                extra=None,
+                verify_hash=False,
+                headers=None
+            )
+            logger.info(f"Successfully uploaded '{object_name}' to container '{self._container_name}'.")
+        except (ObjectError, ContainerDoesNotExistError, InvalidContainerNameError) as e:
+            logger.exception(f"Error uploading the file '{object_name}': {e}")
+        except Exception as e:
+            logger.exception(f"An unexpected error occurred while uploading '{object_name}': {e}")
+
+    def object_exists(self, local_path):
+        """
+        Checks if an object exists in the object storage container.
+
+        Parameters
+        ----------
+        local_path : str
+            The local file path corresponding to the object name.
+
+        Returns
+        -------
+        bool
+            True if the object exists, False otherwise.
+
+        Logs
+        ----
+        Logs information about the existence check or errors encountered.
+        """
+        object_name = os.path.basename(local_path)
+        try:
+            self.driver.get_object(
+                container_name=self._container_name,
+                object_name=object_name
+            )
+            logger.debug(f"Object '{object_name}' exists in container '{self._container_name}'.")
+            return True
+        except ObjectDoesNotExistError:
+            logger.debug(f"Object '{object_name}' does not exist in container '{self._container_name}'.")
+            return False
+        except ContainerDoesNotExistError:
+            logger.error(f"Container '{self._container_name}' does not exist in the cloud storage.")
+            return False
+        except InvalidContainerNameError:
+            logger.error(f"Invalid container name '{self._container_name}'.")
+            return False
+        except Exception as e:
+            logger.exception(f"An unexpected error occurred while checking for object '{object_name}': {e}")
+            return False
```

Comment:
We can remove 4 of returns by moving the return False to the end of the logic

Reply:
Good point, thank you, I was adding exceptions sequentially while reading docs and experimenting with different scenarios, so I end up in this ugly piece of code:D Fixed!